### PR TITLE
Update the Private Data Asset Transfer for Purge

### PR DESCRIPTION
**Pull request by mbwhite**
Add a Purge transaction function, very similar to delete, but calls
Purge instead to completely remove data from the private collections

Signed-off-by: Matthew B White <whitemat@uk.ibm.com>

### CODE REVIEW

**Comment by jkneubuh** [comment]
/azp run

test flake?

File: asset-transfer-private-data/chaincode-go/chaincode/asset_transfer.go [hunk 1]
```diff
@@ -475,6 +475,79 @@ func (s *SmartContract) DeleteAsset(ctx contractapi.TransactionContextInterface)
...
+	valAsbytes, err = ctx.GetStub().GetPrivateData(ownerCollection, assetPurgeInput.ID)
+	if err != nil {
+		return fmt.Errorf("failed to read asset from owner's Collection: %v", err)
+	}
+	if valAsbytes == nil {
+		return fmt.Errorf("asset not found in owner's private Collection %v: %v", ownerCollection, assetPurgeInput.ID)
+	}
+
+	// delete the asset from state
+	err = ctx.GetStub().DelPrivateData(assetCollection, assetPurgeInput.ID)
```
Comment (denyeart) [inline asset-transfer-private-data/chaincode-go/chaincode/asset_transfer.go:536]: I assume any reference to DelPrivateData() should simply be changed to PurgePrivateData()

File: asset-transfer-private-data/chaincode-go/chaincode/asset_transfer.go [hunk 2]
```diff
@@ -475,6 +475,79 @@ func (s *SmartContract) DeleteAsset(ctx contractapi.TransactionContextInterface)
...
+	if err != nil {
+		return fmt.Errorf("failed to delete state: %v", err)
+	}
+
+	// Finally, delete private details of asset
+	err = ctx.GetStub().DelPrivateData(ownerCollection, assetPurgeInput.ID)
```
Comment (denyeart) [inline asset-transfer-private-data/chaincode-go/chaincode/asset_transfer.go:542]: Same here.

File: asset-transfer-private-data/chaincode-go/chaincode/asset_transfer.go [hunk 3]
```diff
@@ -475,6 +475,79 @@ func (s *SmartContract) DeleteAsset(ctx contractapi.TransactionContextInterface)
 
 }
 
+// DeleteAsset can be used by the owner of the asset to delete the asset
```
Comment (denyeart) [inline asset-transfer-private-data/chaincode-go/chaincode/asset_transfer.go:478]: PurgeAsset

**Pull request closed without merging**